wrappers/services.go

```diff
@@ -481,6 +500,15 @@ WantedBy={{.ServicesTarget}}
 		// systemd runs as PID 1 so %h will not work.
 		Home: "/root",
 	}
+	if appInfo.IsUserService() {
+		wrapperData.ServicesTarget = systemd.UserServicesTarget
+		wrapperData.WorkingDir = appInfo.Snap.UserDataDir("%h")
+	} else {
+		wrapperData.ServicesTarget = systemd.ServicesTarget
+		wrapperData.PrerequisiteTarget = systemd.PrerequisiteTarget
+		wrapperData.MountUnit = filepath.Base(systemd.MountUnitPath(appInfo.Snap.MountDir()))
```

Comment:
Why is the mount unit not referenced for user services? Can user services depend on system services somehow?

Reply:
Because it is a system unit, so belongs to a different systemd instance.  It looks like there are synthesised mount units in the user systemd instance, but they end up with names based on the mount point:

    $ systemctl --user list-units | grep skype
    snap-skype-54.mount     loaded active     mounted   /snap/skype/54

Presumably these units would have different names on e.g. Fedora where we mount snaps under `/var/lib/snapd/snap`.